packages/gradle/src/generators/init/init.ts

```diff
@@ -8,13 +8,21 @@ import {
   Tree,
   updateNxJson,
 } from '@nx/devkit';
+import { execSync } from 'child_process';
 import { nxVersion } from '../../utils/versions';
 import { InitGeneratorSchema } from './schema';
 import { hasGradlePlugin } from '../../utils/has-gradle-plugin';
 
 export async function initGenerator(tree: Tree, options: InitGeneratorSchema) {
   const tasks: GeneratorCallback[] = [];
 
+  if (!tree.exists('settings.gradle') && !tree.exists('settings.gradle.kts')) {
+    logger.warn(`Could not find 'settings.gradle' or 'settings.gradle.kts' file in your gradle workspace.
+A Gradle build should contain a 'settings.gradle' or 'settings.gradle.kts' file in its root directory. It may also contain a 'build.gradle' or 'build.gradle.kts' file.
+Running 'gradle init':`);
+    execSync('gradle init', { stdio: 'inherit' });
```

Comment:
Did you do this with an `execSync` here because it would get picked up later on via the Tree? :eyes:

Reply:
yes. just needs to be called before the function addProjectReportToBuildGradle so when checking `if (tree.exists('settings.gradle.kts')) {`, it would be true